react_app/src/actions/index.js

```diff
@@ -6,13 +6,15 @@ export const fetched_courses = courses => {
     };
 };
 
-export const fetchCourses = () => {
+export const fetchCourses = (currentPage) => {
     return function (dispatch, getState) {
-        return fetch(`/courses`)
+        const url = `/courses?page=` + currentPage;
+        return fetch(url)
             .then(
                 data => data.json())
             .then(data => {
-                    dispatch(fetched_courses(data))
+                    dispatch(change_page_count(data['pageCount'])) 
+                    dispatch(fetched_courses(data['data']))
```

Comment:
With the changes from the backend, `data` now looks like `{pageCount: <integer>, data: [<courses1>, <courses2>]}`